OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationWillShowInForegroundManager.java

```diff
@@ -0,0 +1,208 @@
+            // Make sure the key we need to to obtain our notifJob exists in the data input of the Worker
+            if (!getInputData().hasKeyWithValueOfType(EXT_NOTIF_JOB_WORKER_KEY, String.class)) {
+                OneSignal.onesignalLog(OneSignal.LOG_LEVEL.ERROR, "getInputData() does not have the key: " + EXT_NOTIF_JOB_WORKER_KEY + " with class type: String, returning Result.failure for Worker");
+                return Result.failure();
+            }
+
+            // The job key must not be null and a job should exist for the key
+            String extNotifJobKey = getInputData().getString(EXT_NOTIF_JOB_WORKER_KEY);
+            NotificationWillShowInForegroundManager manager = NotificationWillShowInForegroundManager.getInstance();
+            if (extNotifJobKey == null || !manager.hasJob(extNotifJobKey)) {
+                OneSignal.onesignalLog(OneSignal.LOG_LEVEL.ERROR, "extNotifJobKey is null or manager does not have a job with key: " + extNotifJobKey + ", returning Result.failure for Worker");
+                return Result.failure();
+            }
+
+            // Make sure current job obtained is not null
+            ExtNotificationGenerationJob currentExtNotifJob = (ExtNotificationGenerationJob) manager.getJob(extNotifJobKey);
+            if (currentExtNotifJob == null) {
+                OneSignal.onesignalLog(OneSignal.LOG_LEVEL.ERROR, "currentExtNotifJob is null for key: " + extNotifJobKey + ", returning Result.failure for Worker");
+                return Result.failure();
+            }
+
+            // Make sure extNotificationWillShowInForegroundHandler is not null
+            if (OneSignal.extNotificationWillShowInForegroundHandler == null) {
+                OneSignal.onesignalLog(OneSignal.LOG_LEVEL.ERROR, "OneSignal.extNotificationWillShowInForegroundHandler is null, returning Result.failure for Worker");
+                return Result.failure();
+            }
+
+            // Fire the ext handler and report success
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.INFO, "Calling OneSignal.extNotificationWillShowInForegroundHandler.notificationWillShowInForeground(currentExtNotifJob) with OS notification id: " + currentExtNotifJob.getApiNotificationId());
+            OneSignal.extNotificationWillShowInForegroundHandler.notificationWillShowInForeground(currentExtNotifJob);
+
+            return Result.success();
+        }
+    }
+
+    public static class AppNotificationWillShowInForegroundWorker extends Worker {
+
+        public AppNotificationWillShowInForegroundWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
+            super(context, workerParams);
+        }
+
+        @NonNull
+        @Override
+        public Result doWork() {
+            OneSignal.onesignalLog(OneSignal.LOG_LEVEL.INFO, "doWork() called for AppNotificationWillShowInForegroundWorker");
+
+            // appContext my have been used to begin "doWork"
+            if (OneSignal.appContext == null) {
```

Comment:
We can remove null check, we will always have a context here.

Reply:
Removing!